### PR TITLE
Users can now visit the forums page to see all existing forums

### PR DESCRIPTION
**Pull request by gabbytang**
## Trello ticket link

[As a user, I want to be able view forums created by other users.](https://trello.com/c/zmjjt7A5 )

## Implementation description
* Created Forums page and added link to navigation bar
* Page shows all the information for existing forums in the database
* Created My Forums basic page and added link to navigation bar 
* Page shows title, description, and date for all forums created by the user 

## Steps to test
1. Make sure the Forums page loads 
2. You can see all information for all existing forums 
3. Make sure the My Forums page loads 
4. Ensure you can see information for all existing forums created by you. 

## What should reviewers focus on?
* Please check that recently added forums are loaded on the Forums page

### CODE REVIEW

File: src/components/Forums.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    return (
+        <div id="body">
+
+            <NavigationBar></NavigationBar>
+
+            <Typography
+                variant="h4"
+                gutterBottom
+                component="div">
+                Forums
```
Comment (Eshah7) [inline src/components/Forums.js]: Not sure if this is an issue I am seeing cause my laptop screen is a bit small - 

However, the Forums title is hiding underneath the NavBar. Could you possibly use `<Box> </Box>`  (used in the ProfileDashboard) to move the Forums title below the NavBar?

File: src/components/Forums.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            </Typography>
+
+            {forums.map((event) => (
```
Comment (Eshah7) [inline src/components/Forums.js:101]: Maybe we could add some space in between each forum using this: <Box sx={{ m: 2 }} /> after the creation of the first forum? 

Unless we are staying consistent with Events!

**Comment by Eshah7** [comment]
Screenshot 2023-02-22 at 7 17 44 PM

Passed the test! I can see the existing forums, thank you!

**Review by Eshah7 — APPROVED** [review]
LGMT!!! Thanks Gabby!

File: routes/forums/getForumsByUser.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+const express = require('express');
+const router = express.Router();
+const mysql = require('mysql');
+const config = require('../../config.js');
+
+//retrives all forum data created by the given user id
+router.post('/api/getEventsByUser', (req, res) => {
```
Comment (safchow) [inline routes/forums/getForumsByUser.js]: endpoint name is incorrect: should be `/api/getForumsByUserID`

File: src/components/Forums.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+            <Box sx={{ position: 'absolute', top: 150, left: "35%" }}>
+            {forums.map((event) => (
+                <Card style={{ width: '500px', marginBottom: '20px' }} key={event.id}>
+                    <CardContent>
+                        <Typography variant="h5" component="div">
+                            {event.forumTitle}<br />
+                        </Typography>
+                        <Typography sx={{ mb: 1.5 }} color="text.secondary">
+                            Posted on {new Date(event.dateTime).toLocaleDateString()} by {event.creatorName}<br />
```
Comment (safchow) [inline src/components/Forums.js]: Is the date rendering the timezone in GMT? If so, this code should convert the date to EST:
```
new Date(new Date(event.dateTime).getTime() - (5 * 60 * 60 * 1000)).toLocaleDateString()
```

Comment (safchow) [inline src/components/Forums.js]: Thoughts on rendering date created and the author on two separate lines?

File: src/components/Forums.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+                    variant="h4"
+                    gutterBottom
+                    component="div">
+                    Forums
+                </Typography>
+            </Box>
+
+            <Box sx={{ position: 'absolute', top: 150, left: "35%" }}>
+            {forums.map((event) => (
+                <Card style={{ width: '500px', marginBottom: '20px' }} key={event.id}>
```
Comment (safchow) [inline src/components/Forums.js]: what are your thoughts on doubling the width of the forums? Try the following and implement it if you think its better:
```
    return (
        <div id="body">

            <NavigationBar></NavigationBar>

            <Box sx={{ position: 'absolute', top: 110, left: '50%', transform: 'translate(-50%, -50%)' }}>
                ....
            </Box>

            <Box sx={{ position: 'absolute', top: 150, left: '50%', transform: 'translateX(-50%)' }}>
                 {forums.map((event) => (
                    <Card style={{ width: '1000px', marginBottom: '20px' }} key={event.id}>
                    ....
            </Box>
```

the first box component changes update the title of the page to be neatly centred on the page and in the correct position horizontally. 

the second box component stretches the width to 1000px and centres them on the page. 

The results look like so:
image

**Review by safchow — CHANGES_REQUESTED** [review]

File: src/components/Forums.js (hunk 4 above)
Comment (gabbytang) [reply src/components/Forums.js:108]: RIght now, the date times are being generated in SQL directly.

File: src/components/Forums.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+            <Box sx={{ position: 'absolute', top: 150, left: '50%', transform: 'translateX(-50%)' }}>
+            {forums.map((event) => (
+                <Card style={{ width: '800px', marginBottom: '20px' }} key={event.id}>
+                    <CardContent>
+                        <Typography variant="h5" component="div">
+                            {event.forumTitle}<br />
+                        </Typography>
+                        <Typography sx={{ mb: 1.5 }} color="text.secondary">
+                            Posted on {new Date(new Date(event.dateTime).getTime() - (5 * 60 * 60 * 1000)).toLocaleDateString()}<br/>
+                            &nbsp; by {event.creatorName}<br />
```
Comment (safchow) [inline src/components/Forums.js:109]: is this extra space intentional? If not, remove the `&nbsp; ` part

**Review by safchow — APPROVED** [review]
lgtm 💯

File: src/components/Forums.js (hunk 6 above)
Comment (gabbytang) [reply src/components/Forums.js:109]: Yeah its intentional, I think it looks a bit better

**Pull request merged**